### PR TITLE
GIX-1328 Only select selectable universes.

### PR DESCRIPTION
**Pull request by dskloetd**
# Motivation

It's currently possible to see uncommitted sns neurons by deep linking the root canister id.
The issue seems to be that while `selectedUniverseStore` only gives a universe from `selectableUniversesStore`, `selectedUniverseIdStore` doesn't care whether the universe is in `selectableUniversesStore`.
The solution is to make both respect selectableUniversesStore by having one derive from the other.
As a side effect we no longer need to check if the selected universe is the ckBTC universe because selectableUniversesStore already does this check.

# Changes

frontend/src/lib/derived/selected-universe.derived.ts: Make sure selectedUniverseIdStore only holds a universe id from selectableUniversesStore.

Many tests had to change because they didn't set up the selectableUniversesStore to allow the selected universe they need for the test.

# Tests

1. Set up selectableUniversesStore correctly via snsQueryStore in many tests.
2. In selected-universe.derived.spec.ts, explicitly test that a universe has to be selectable to be selected.
3. Made sure the new test fails without the fix from this PR.

### CODE REVIEW

File: frontend/src/lib/derived/selected-universe.derived.ts [hunk 1]
```diff
@@ -39,19 +34,19 @@ const pageUniverseIdStore: Readable<Principal> = derived(
...
+
 export const selectedUniverseIdStore: Readable<Principal> = derived<
-  [Readable<Principal>, Readable<Page>, Readable<boolean>],
+  Readable<Universe>,
   Principal
->(
-  [pageUniverseIdStore, pageStore, ENABLE_CKBTC_LEDGER],
-  ([canisterId, page, enableCkbtcLedger]) => {
-    // ckBTC is only available on Accounts therefore we fallback to Nns if selected and user switch to another view
-    if (enableCkbtcLedger && pathSupportsCkBTC(page)) {
```
Comment (lmuntaner) [inline frontend/src/lib/derived/selected-universe.derived.ts:49]: Do you cover the case where the path doesn't support ckbtc, but the `u` is set to ckBTC universe?

What should we do in that case?

This might happen if you are in Accounts, click on ckBTC, and then move to Neurons. The `u` selected might still be `ckBTC`. Therefore, neurons won't be found or an error will be raised.

File: frontend/src/tests/lib/derived/selected-universe.derived.spec.ts [hunk 2]
```diff
@@ -99,5 +116,19 @@ describe("selected universe derived stores", () => {
       const $store2 = get(selectedUniverseIdStore);
       expect($store2.toText()).toEqual(OWN_CANISTER_ID.toText());
     });
+
+    it("returns OWN_CANISTER_ID if context is not a selectable universe", () => {
```
Comment (lmuntaner) [inline frontend/src/tests/lib/derived/selected-universe.derived.spec.ts:120]: Should we also add a test regarding the edge case of the path not supporting `ckBTC` but the `u` being ckBTC?

**Review by lmuntaner — COMMENTED** [review]
Looking good, but there is an edge case that I'm not sure how it behaves and how we should treat it.

Let me know whether you encountered it and what happened.

File: frontend/src/tests/lib/derived/selected-universe.derived.spec.ts (hunk 2 above)
Comment (dskloetd) [reply frontend/src/tests/lib/derived/selected-universe.derived.spec.ts:135]: Since that's existing functionality that I didn't change (just relied on a different mechanism to get the same functionality), I assumed it would be covered by the existing ckBTC tests in this file. But on a close look it seems like it wasn't covered. So I've added the missing tests. Thanks!

**Review by peterpeterparker — CHANGES_REQUESTED** [review]
Tested it locally. This PR does not list ckBTC anymore.

Capture d’écran 2023-02-22 à 10 38 32
isn't ok.

**Comment by peterpeterparker** [comment]
(double checking...)

**Comment by peterpeterparker** [comment]
Sounds good. ckBTC is listed including correct balance.
Capture d’écran 2023-02-22 à 10 53 53
Capture d’écran 2023-02-22 à 10 53 55

**Review by peterpeterparker — APPROVED** [review]
LGTM according my test, thanks.

**Comment by dskloetd** [comment]
> LGTM according my test, thanks.

Thanks for verifying!

File: frontend/src/tests/lib/derived/selected-universe.derived.spec.ts (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/derived/selected-universe.derived.spec.ts:135]: thanks!

File: frontend/src/lib/derived/selected-universe.derived.ts (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/lib/derived/selected-universe.derived.ts:49]: Ah!! Thanks, I forgot about that. Great, thanks for the explanation!
